pkg/ddb/builder_query.go

```diff
@@ -70,6 +72,7 @@ func NewQueryBuilder(metadata *Metadata) QueryBuilder {
 	return &queryBuilder{
 		metadata: metadata,
 		selected: metadata.Main,
+		clock:    clock.Provider,
```

Comment:
Does it make sense to have the clock as a parameter in `NewQueryBuilder`?  
Basically deciding what clock to use on repository creation?